includes/classes/shopping_cart.php

```diff
@@ -1808,6 +1800,13 @@ function actionUpdateProduct($goto, $parameters) {
           default:
             $adjust_max= 'false';
           }
+          
+// bof: notify about adjustment to new quantity to be same as current in stock or maximum to add
+          if ($adjust_max == 'true') {
+            $messageStack->add_session('shopping_cart', ($_SESSION['cart']->display_debug_messages ? 'FUNCTION ' . __FUNCTION__ . ': ' : '') . WARNING_PRODUCT_QUANTITY_ADJUSTED . zen_get_products_name($_POST['products_id'][$i]), 'caution');
```

Comment:
Curious: Why did you change to using `$_SESSION['cart']->display_debug_messages` here, when `$this` already refers to `$_SESSION['cart']` because it's the self-same object